falcon/testing/helpers.py

```diff
@@ -631,6 +631,12 @@ def create_environ(path='/', query_string='', http_version='1.1',
     if content_length != 0:
         env['CONTENT_LENGTH'] = str(content_length)
 
+    if cookies is not None and method != 'OPTIONS':
```

Comment:
why the filter on the method?

Reply:
`OPTIONS` method discards `Set-Cookie` header, and that's the reason why we are checking the method here.

Reply:
Should we then add a `NOTE` explaining this for the source code reader?

Reply:
IMHO a `NOTE` would be helpful.